pkg/che/che.go

```diff
@@ -23,7 +40,123 @@ type Client struct {
 func InitDefaultCheClient(cfg *crtcfg.Config, cl client.Client) {
 	DefaultClient = &Client{
 		config:     cfg,
+		httpClient: newHTTPClient(),
 		k8sClient:  cl,
 		tokenCache: newTokenCache(),
 	}
 }
+
+// UserExists returns true if the username exists, false if it doesn't and an error if there was problem with the request
+func (c *Client) UserExists(username string) (bool, error) {
+	reqData := url.Values{}
+	reqData.Set("name", username)
+	res, err := c.cheRequest(http.MethodGet, cheUserFindPath, reqData)
+	if err != nil {
+		return false, errors.Wrapf(err, "request to find Che user '%s' failed", username)
+	}
+	defer rest.CloseResponse(res)
+	if res.StatusCode == http.StatusOK {
+		return true, nil
+	} else if res.StatusCode == http.StatusNotFound {
+		return false, nil
+	}
+	return false, errors.Errorf("request to find Che user '%s' failed, Response status: '%s'", username, res.Status)
+}
+
+// GetUserIDByUsername returns the user ID that maps to the given username
+func (c *Client) GetUserIDByUsername(username string) (string, error) {
```

Comment:
It will return an error if the user is not found, right? Let's add it to the function description.
Is the plan to use UserExists() first and if it does exist then use GetuserIDByUsername?

Reply:
Right, it will return an error. I've updated the comment to explain that.

The planned usage can be seen here: https://github.com/rajivnathan/member-operator/blob/64cf4a336a8dcf416bf8ce6eee937b52b9a85c26/pkg/controller/useraccount/useraccount_controller.go#L653